app/models/sitemap.rb

```diff
@@ -0,0 +1,107 @@
+require "builder"
+
+class Sitemap
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+
+  include Sitemap::RobotsUpdatable
+
+  ERROR_PAGES = %w[400.html 404.html 406-unsupported-browser.html 422.html 500.html].freeze
+
+  attribute :base_url, :string
+  attribute :build_dir, :string, default: -> { Parklife.application.config.build_dir }
+  attribute :update_robots, :boolean, default: false
+
+  validates :base_url, presence: true, format: { with: URI.regexp(%w[http https]) }
+
+  def generate!
+    write_sitemap!
+    write_compressed_sitemap!
+
+    handle_robots_update!
+
+    total_pages
+  end
+
+  private
+
+  # File writing
+  def write_sitemap!
+    File.write(sitemap_path, xml_sitemap)
+
+    puts "Generated sitemap.xml with #{total_pages} URLs"
+  end
+
+  def write_compressed_sitemap!
+    Zlib::GzipWriter.open(sitemap_path(compressed: true)) do |gzip|
+      gzip.write(xml_sitemap.squish)
+    end
+
+    puts "Generated sitemap.xml.gz (compressed)"
+  end
+
+  # XML generation
+  def xml_sitemap
```

Comment:
Excellent use of memoization here\! This ensures the XML is only generated once and reused for both the regular and compressed versions.